javalanglib/src/main/scala/java/lang/Math.scala

```diff
@@ -305,13 +305,35 @@ object Math {
     if (a >= Integer.MIN_VALUE && a <= Integer.MAX_VALUE) a.toInt
     else throw new ArithmeticException("Integer overflow")
 
+  def floorDiv(a: scala.Int, b: scala.Int): scala.Int = {
+    if (b == 0)
+      throw new ArithmeticException("/ by 0")
```

Comment:
Yes, I know. When the JavaDoc says it throws a kind of exception which is an undefined behavior, it's always a judgment call whether to check and throw or not. Sometimes it's hard to decide, but here it's pretty clear that the method should behave the same way as the "non-floor" variant, i.e., the primitive operator. So it shouldn't explicitly check.